pkg/sdk/sequences_def.go

```diff
@@ -0,0 +1,115 @@
+package sdk
+
+import g "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk/poc/generator"
+
+//go:generate go run ./poc/main.go
+
+var sequenceSet = g.NewQueryStruct("SequenceSet").
+	PredefinedQueryStructField("ValuesBehavior", "*ValuesBehavior", g.KeywordOptions()).
+	OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes())
+
+var SequencesDef = g.NewInterface(
+	"Sequences",
+	"Sequence",
+	g.KindOfT[SchemaObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/create-sequence",
+	g.NewQueryStruct("CreateSequence").
+		Create().
+		OrReplace().
+		SQL("SEQUENCE").
+		IfNotExists().
+		Name().
+		OptionalSQL("WITH").
+		OptionalNumberAssignment("START", g.ParameterOptions().NoQuotes()).
```

Comment:
Looking at this I realized we already have a similar (or even exact?) object `ColumnIdentity`:
```
type ColumnIdentity struct {
	Start     int   `ddl:"parameter,no_quotes,no_equals" sql:"START"`
	Increment int   `ddl:"parameter,no_quotes,no_equals" sql:"INCREMENT"`
	Order     *bool `ddl:"keyword" sql:"ORDER"`
	Noorder   *bool `ddl:"keyword" sql:"NOORDER"`
}
```
I am not pushing hard for reusing it here but we were reusing similar structures for other objects.

Maybe lets leave a TODO here and add it as a subtask to SNOW-933289? WDYT @sfc-gh-swinkler @sfc-gh-jcieslak ?

Reply:
Created Jira SNOW-1016272